internal/stacks/stackconfig/provider_requirements.go

```diff
@@ -201,6 +202,13 @@ func decodeProviderRequirementsBlock(block *hcl.Block) (*ProviderRequirements, t
 
 func (pr *ProviderRequirements) ProviderForLocalName(localName string) (addrs.Provider, bool) {
 	if pr == nil {
+		// Check whether the provider *might be* built-in.
+		// There might be special occasions where localName corresponds to a
+		// built-in provider names. In that case, customers will see an error
+		// occurring later in the process.
+		if _, ok := builtinProviders.BuiltInProviders()[localName]; ok {
+			return addrs.Provider{}, true
```

Comment:
I think there's like an `addrs.BuiltinProvider(localName)` function you can call to return a proper address here?